src/main/java/stringcalculator/StringCalculator.java

```diff
@@ -0,0 +1,63 @@
+package stringcalculator;
+
+import org.apache.logging.log4j.util.Strings;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class StringCalculator {
+    private final Pattern customSeparatorPattern = Pattern.compile("//(.+?)\\n(.*)");
+    private final List<String> defaultSeparators = List.of(":", ",");
+
+    public StringCalculator() {
+    }
+
+    public Integer calculate(String input) {
+        if (Strings.isEmpty(input)) {
+            return 0;
+        }
+
+        List<String> separators = new ArrayList<>(this.defaultSeparators);
+        SeparatorParser separatorParser = new SeparatorParser(this.customSeparatorPattern);
+        CustomSeparator customSeparator = separatorParser.parseSeparator(input);
+
+        if (customSeparator.isParsedStatus()) {
+            input = customSeparator.getParsedInput();
+            separators.add(customSeparator.getSeparator());
+        }
+
+        String[] splitInputNumbers = input.split(getJoinedSeparatorsString(separators));
```

Comment:
파싱하는 책임을 분리해볼 수도 있겠어요!

```suggestion
String[] splitInputNumbers = new NumbersParser().parse(input);
```